iam/policy/policy.go

```diff
@@ -223,10 +223,14 @@ func (iamp Policy) Merge(input Policy) Policy {
 		mergedPolicy.Version = input.Version
 	}
 	for _, st := range iamp.Statements {
-		mergedPolicy.Statements = append(mergedPolicy.Statements, st.Clone())
+		var newSt Statement = st.Clone()
+		newSt.SID = st.SID
+		mergedPolicy.Statements = append(mergedPolicy.Statements, newSt)
 	}
 	for _, st := range input.Statements {
-		mergedPolicy.Statements = append(mergedPolicy.Statements, st.Clone())
+		var newSt Statement = st.Clone()
+		newSt.SID = st.SID
+		mergedPolicy.Statements = append(mergedPolicy.Statements, newSt)
```

Comment:
It is really debatable which Sid is valid here - my assumption here is that we should preserve the Sid that is `inline` instead of blindly replacing it with any Sid that comes along.

This type of replacement has a bug where it will still become empty if you have other Statements without Sid.

Reply:
@harshavardhana Thanks for checking my PR

I'm not sure that by `inline` you mean the policy that user gives us while creating a new Service Account?

If that's the case `input` comes later in `mergedPolicy` and overriders previous ones in dropping duplicate statements process.

About the second concern, I think you mean if two statements are same but the second one has no Sid, it will override the one that has. On that matter, maybe we could always pass the Sid forward when it comes to that situation, something like this:

```go
func (iamp *Policy) dropDuplicateStatements() {
redo:
	for i := range iamp.Statements {
		for j, statement := range iamp.Statements[i+1:] {
			if !iamp.Statements[i].Equals(statement) {
				continue
			}
			if iamp.Statements[i+1+j].SID == "" && iamp.Statements[i].SID != "" {
				iamp.Statements[i+1+j].SID = iamp.Statements[i].SID
			}
			iamp.Statements = append(iamp.Statements[:i], iamp.Statements[i+1:]...)
			goto redo
		}
	}
}
```

Pardon me if my understanding is wrong in any of these, since I have just recently started using minio and have very small knowledge on it's source code 🙏

Reply:
The bug is actually not here in preserving the `Sid` instead the way service account was returning an incorrect Policy by doing an unnecessary Merge.